fhir-persistence-params/src/main/java/com/ibm/fhir/persistence/params/batch/ParameterValueCollector.java

```diff
@@ -0,0 +1,364 @@
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Get the LogicalResourceIdentValue we've assigned for the given (resourceType, logicalId)
+     * tuple. The returned value may not yet have the actual logical_resource_id yet - we fetch
+     * these values later and create new database records as necessary
+     * @param resourceType
+     * @param logicalId
+     * @return
+     */
+    private LogicalResourceIdentValue lookupLogicalResourceIdentValue(String resourceType, String logicalId) {
+        LogicalResourceIdentKey key = new LogicalResourceIdentKey(resourceType, logicalId);
+        LogicalResourceIdentValue result = this.logicalResourceIdentMap.get(key);
+        if (result == null) {
+            result = LogicalResourceIdentValue.builder()
+                    .withResourceTypeId(identityCache.getResourceTypeId(resourceType))
+                    .withResourceType(resourceType)
+                    .withLogicalId(logicalId)
+                    .build();
+            this.logicalResourceIdentMap.put(key, result);
+
+            // see if we can find the logical_resource_id from the cache
+            Long logicalResourceId = identityCache.getLogicalResourceIdentId(resourceType, logicalId);
+            if (logicalResourceId != null) {
+                result.setLogicalResourceId(logicalResourceId);
+            } else {
+                // Add to the unresolved list to look up later
+                this.unresolvedLogicalResourceIdents.add(result);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Get the CommonCanonicalValue we've assigned for the given url value.
+     * The returned value may not yet have the actual canonical_id yet - we fetch
+     * these values later and create new database records as necessary.
+     * @param url
+     * @return
+     */
+    private CommonCanonicalValue lookupCommonCanonicalValue(String url) {
+        CommonCanonicalValueKey key = new CommonCanonicalValueKey(FIXED_SHARD, url);
+        CommonCanonicalValue result = this.commonCanonicalValueMap.get(key);
+        if (result == null) {
+            result = new CommonCanonicalValue(FIXED_SHARD, url);
+            this.commonCanonicalValueMap.put(key, result);
+
+            // Take this opportunity to see if we have a cached value for this common token value
+            Long canonicalId = identityCache.getCommonCanonicalValueId(FIXED_SHARD, url);
+            if (canonicalId != null) {
+                result.setCanonicalId(canonicalId);
+            } else {
+                this.unresolvedCanonicalValues.add(result);
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public void publish(IParamValueProcessor processor) throws FHIRPersistenceException {
+        // resolve any missing ids
+        try (MetricHandle createMetric = FHIRRequestContext.get().getMetricHandle(ParamMetrics.M_PARAM_RESOLVE_LOGICAL_RESOURCES.name())) {
```

Comment:
more cases of the `createMetric` variable name.  can't remember if you changed those other ones or kept them as `createMetric`, but the same comment applies (could be short name like `mh` and have about the same value semantically)

Reply:
also, I think it would be safe to just get FHIRRequestContext once and then use that instance for all these getMetricHandle calls.  pretty minor though.

Reply:
I changed them. Will change this too. Yes I toyed with getting the request context once, but the other half of my brain likes the self-contained nature of the lines.